ghcide/src/Development/IDE/Plugin/HLS.hs

```diff
@@ -58,8 +58,8 @@ asGhcIdePlugin :: Recorder (WithPriority Log) -> IdePlugins IdeState -> Plugin C
 asGhcIdePlugin recorder (IdePlugins ls) =
     mkPlugin rulesPlugins HLS.pluginRules <>
     mkPlugin executeCommandPlugins HLS.pluginCommands <>
-    mkPlugin extensiblePlugins HLS.pluginHandlers <>
-    mkPlugin (extensibleNotificationPlugins recorder) HLS.pluginNotificationHandlers <>
+    mkPlugin extensiblePlugins id <>
+    mkPlugin (extensibleNotificationPlugins recorder) id <>
```

Comment:
I am not entirely sure... 

I don't think it is sensible, the replacement would be:

```haskell
    extensiblePlugins recorder ls <>
```

but according to the documentation of `mkPlugin`, this'd need to be rather:
```haskell
    (if null ls then mempty else extensiblePlugins recorder ls) <>
``` 
or something identical.

Reply:
I think I just don't understand how these lines work, never mind.

Reply:
actually, this looks like a very slightly subtle change we introduced here

Reply:
I want to restore the previous behaviour before merging

Reply:
Essentially, we need to find the PluginDescriptor for a given PluginId, because 'IdeHandler' only gives us 
```haskell
fs' :: [(PluginId,
  IdeState
  -> MessageParams a
  -> LspM
       Config (NonEmpty (Either ResponseError (ResponseResult a))))] 
```

That's why we look up the plid in the list of plugins that we know about, but since this unsafe (e.g. maybe the plid doesn't exist in the list of plugin descriptors that we have?) we have `[(PluginId, Maybe PluginDescriptor)]`. So, we clean this up and make this into `[(PluginId, PluginDescriptor)]` (via `collectPluginInfo`, maybe misnomer?). Afterwards, we can finally check whether this Handler of the plugin `plid` is really enabled for the given request. If so, run all the enabled handlers just as the usual. 
The additional code is only to make sure we obtain the `PluginDescriptor` given a plid safely and report an error if there is an internal inconsistency.

I am not sure how to properly refactor this, maybe `IdeHandler` should rather hold a `PluginDescriptor` instead of a `PluginId`, then we wouldn't need this round-tripping.

Reply:
This did not introduce a change by accident, so it is safe to go. Maybe we can still refactor it?

Reply:
Refactored, please let me know what you think